scripts/main.js

```diff
@@ -88,4 +94,30 @@ const closePopUp = () => {
 popUpClose.addEventListener('click', closePopUp);
 btn1.addEventListener('click', btn1Display);
 btn2.addEventListener('click', btn2Display);
-btn3.addEventListener('click', btn3Display);
\ No newline at end of file
+btn3.addEventListener('click', btn3Display);
+
+// Validate forms
+formSpace.addEventListener('submit', (event) => {
+  if (Caps.test(Useremail.value)) {
+    event.preventDefault();
+    formError.innerHTML = 'Email must not contain capital letters....!';
+  } else {
+    formError.innerHTML = '';
+  }
+});
+
+// Local Storage
+formSpace.addEventListener('input', () => {
+  const collectedInput = {
+    name: Username.value,
+    email: Useremail.value,
+    message: Usermessage.value,
+  };
+
+  localStorage.setItem('collectedInput', JSON.stringify(collectedInput));
+});
+
+const collectedInputRetrive = JSON.parse(localStorage.getItem('collectedInput'));
+Username.value = collectedInputRetrive.name;
+Useremail.value = collectedInputRetrive.email;
+Usermessage.value = collectedInputRetrive.message;
```

Comment:
- [ ] Great job so far :clap:  However When the user loads the page, the code checks if there are any saved data in the local storage and if so the input fields are pre-filled with the data, kindly make this works as required

eg:

```
window.onload = () => {
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
};
```
- [ ] When the user changes the content of any input field in the form the data object is updated and is saved to the local storage so as when the user loads the page, the input fields are pre-filled with updated data